SwanProjects/swanprojects/handlers.py

```diff
@@ -0,0 +1,363 @@
+            self.swan_config.projects_folder_name, name)
+
+        try:
+            self.contents_manager._save_project(project_relative_dir, None)
+        except Exception as msg:
+            data = {"status": False, "project_dir": project_relative_dir,
+                    "msg": f"Error creating folder for project {name}, traceback: {msg}"}
+            self.finish(json.dumps(data))
+            return
+        swan_project_content = {'stack': stack, 'release': release,
+                                'platform': platform}
+
+        if not self.save_project_file(project_relative_dir, swan_project_content):
+            return
+
+        try:
+            swan_user_script_file = os.path.join(
+                project_relative_dir, self.swan_config.userscript_file_name)
+            self.contents_manager.new(
+                {'type': 'file', 'content': user_script, 'format': 'text'}, swan_user_script_file)
+        except Exception as msg:
+            data = {"status": False, "project_dir": project_relative_dir,
+                    "msg": f"Error creating {self.swan_config.userscript_file_name} file for project {name}, traceback: {msg}"}
+            self.finish(json.dumps(data))
+            return
+
+        km_info = self.get_kmanager_info(project_relative_dir)
+        swan_project_content.update(km_info)
+
+        if not self.save_project_file(project_relative_dir, swan_project_content):
+            return
+
+        data = {"status": True, "project_dir": project_relative_dir,
+                "msg": f"created project {name}"}
+        self.finish(json.dumps(data))
+
+
+class EditProjectHandler(SwanAPIHandler):
+
+    @tornado.web.authenticated
+    def put(self):
+        """
+        This endpoint allows to edit project information, such as name, stack, platform etc..
+        The project can be renamed from $HOME/SWAN_projects/old_name to $HOME/SWAN_projects/name
+        and metadata in the .swanproject is updated.
+
+        This endpoint allows also edit corrupted projects, if something is wrong in .swanproject
+        the edit project dialog will send the information again to this endpoint to fix the project information.
+        """
+        input_data = self.get_json_body()
+        self.log.info(f"EditProjectHandler = {input_data}")
+
+        corrupted = input_data.get("corrupted")
+        old_name = input_data.get("old_name")
+        old_stack = input_data.get("old_stack")
+        old_platform = input_data.get("old_platform")
+        old_release = input_data.get("old_release")
+        old_userscript = input_data.get("old_userscript")
+
+        name = input_data["name"]
+        stack = input_data["stack"]
+        platform = input_data["platform"]
+        release = input_data["release"]
+        user_script = input_data["user_script"]
+
+        project_relative_dir = os.path.join(
+            self.swan_config.projects_folder_name, name)
+        if old_name != name:
```

Comment:
`old_name` can be `None`, right? Be careful below when you use it.

Check also the other variables you obtain via `input_data.get`, if they can be `None` and whether this affects how you use them below.

Reply:
fixed, I am checking it for rename, for the others conditionals old_* can be None without problem.
I am using the new values to do the procedures, but for rename I need the old_name not None.
Thanks for this catch!